EIPS/eip-7937.md

```diff
@@ -0,0 +1,145 @@
+---
+eip: 7937
+title: 64-bit mode EVM operations
+description: Multibyte opcodes for 64-bit arithmetic, comparison, bitwise, stack and flow operations in EVM.
+author: Wei Tang (@sorpaas)
+discussions-to: https://ethereum-magicians.org/t/eip-9687-64-bit-mode-evm-operations/23794
+status: Draft
+type: Standards Track
+category: Core
+created: 2025-04-23
+---
+
+## Abstract
+
+This EIP introduces multibyte opcodes prefixed by `C0` for 64-bit arithmetic (`C001`-`C00B`), comparison (`C010`-`C015`), bitwise (`C016`-`C019`), stack (`DUP*` and `SWAP*`) and flow (`C056` and `C057` in "legacy", or `C0E1` and `C0E2` in EOFv1) operations.
```

Comment:
Note that these multibyte opcodes are expensive to deposit. Each byte costs 200 gas to deposit. So although this bytecode layout is nice, it will mean that users will pay more to deposit this code. Also note that there is a limit of size of a smart contract, so if someone would use mostly these 64-bit opcodes in a contract that would thus mean the maximum size of that contract is effectively halved (half of operations are available when compared to non-64-bit opcodes)

Reply:
Aha, but I did not get this immediately (might be something to add for clarity): the "normal" opcodes are re-used here, but if these are thus prefixed by the `C0` byte they run in 64-bit mode. So this `C0` is basically an "optional opcode flag". This is very creative, I like it, but please note that this will cost more to deposit, and will also limit the total program size (doubles cost, halves code size, in the extreme cases of only using 64bit).

Since it is likely (?) that multiple 64-bit operations are chained after each other, would you consider a variant where one can set the 64-bit mode on and off? (Either two opcodes, or one opcode which pops from stack?)